src/openfermion/measurements/_time_series_estimators.py

```diff
@@ -0,0 +1,460 @@
+            automatic_average: whether to reestimate the eigenvalues whenever
+                new data is added.
+        '''
+        if num_freqs_max * (1+depolarizing_noise) > max_experiment_length:
+            raise ValueError('''Maximum experiment length of {}
+                             is too short to estimate {} frequencies'''.format(
+                                 max_experiment_length, num_freqs_max))
+
+        self._max_experiment_length = max_experiment_length
+        self._num_freqs_max = num_freqs_max
+        self._depolarizing_noise = depolarizing_noise
+        self._singular_values_cutoff = singular_values_cutoff
+        self._automatic_average = automatic_average
+        self._store_history = store_history
+
+        self.reset()
+
+    def reset(self):
+        '''
+        Resets estimator for new experiment
+        '''
+        self._re_abs = np.zeros(self._max_experiment_length+1)
+        self._im_abs = np.zeros(self._max_experiment_length+1)
+        self._re_est = np.zeros(self._max_experiment_length+1)
+        self._im_est = np.zeros(self._max_experiment_length+1)
+        self._function_estimate = np.zeros(self._max_experiment_length+1)
+        self._function_estimate[0] = 1  # we get this point for free.
+        self._num_freqs = self._num_freqs_max
+        self._update_flag = True
+
+        if self._store_history:
+            self.average_history = []
+
+    def update(self, experiment_data):
+        '''
+        Takes new experiment data in dictionary format. Note - storing data
+        in this format is significantly worse than tallying results and
+        inserting with update_mass (provided here to work similarly
+        to the Bayesian estimators).
+
+        Args:
+            experiment_data: list of dictionaries containing information
+                about each round in the experiment. Each dictionary
+                must contain the entries 'final_rotation', 'measurement',
+                and 'num_rotations'. For the single-round estimator,
+                experiment_data must be of length 1.
+
+        Returns:
+            success of update (always True for this estimator)
+        '''
+        if len(experiment_data) > 1:
+            raise NotImplementedError('''Estimator not implemented for
+                                      multi-round experiments''')
+        round_data = experiment_data[0]
+        angle = round_data['final_rotation'] +\
+            np.pi * round_data['measurement']
+        k = round_data['num_rotations']
+
+        self._re_abs[k] += np.cos(angle)**2
+        self._im_abs[k] += np.sin(angle)**2
+        self._re_est[k] += np.cos(angle)
+        self._im_est[k] -= np.sin(angle)
+
+        self._update_flag = False
+
+        if self._automatic_average is True:
+            self.estimate()
+
+        return True
+
+    def update_mass(self, f_data):
+        '''
+        Takes data to update g(k) in the form  The preferable form to input
```

Comment:
`in the form` ... ?

Reply:
Good catch, I'd left this hanging and described a few lines below instead (with the definition of f_data). I've just removed this comment now.